### PR TITLE
Fix: Failing test in data_modules(dp)

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fix: Failing test in data_modules(dp)

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SeanNaren — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Comment by justusschock** [comment]
@kaushikb11 can you do the same for https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/models/test_restore.py#L211 thanks :)

**Review by Borda — APPROVED** [review]

**Comment by kaushikb11** [comment]
Hi @williamFalcon, could you review this?

File: tests/core/test_datamodules.py [hunk 1]
```diff
@@ -381,8 +381,8 @@ def _step(self, batch, batch_idx):
         def training_step(self, batch, batch_idx):
             _, y = batch
             out = self._step(batch, batch_idx)
-            out['loss'] = F.cross_entropy(out['logits'], y)
-            return out
+            loss = F.cross_entropy(out['logits'], y)
+            return loss
```
Comment (rohitgr7) [inline tests/core/test_datamodules.py:385]: just wondering why returning a dict doesn't work with dp. can you tell?

Comment (Borda) [reply tests/core/test_datamodules.py:385]: discs shall be also supported... @SeanNaren?

**Review by williamFalcon — APPROVED** [review]

File: tests/core/test_datamodules.py (hunk 1 above)
Comment (rohitgr7) [reply tests/core/test_datamodules.py:385]: I think it should else it would be impossible to use metrics from PL while training in DP mode.
